lightning/src/ln/peer_handler.rs

```diff
@@ -425,7 +420,25 @@ pub struct PeerManager<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: De
 		L::Target: Logger,
 		CMH::Target: CustomMessageHandler {
 	message_handler: MessageHandler<CM, RM>,
-	peers: Mutex<PeerHolder<Descriptor>>,
+	/// Peer is under its own mutex for sending and receiving bytes, but note that we do *not* hold
+	/// this mutex while we're processing a message. This is fine as [`PeerManager::read_event`]
+	/// requires that there be no parallel calls for a given peer, so mutual exclusion of messages
+	/// handed to the `MessageHandler`s for a given peer is already guaranteed.
+	peers: FairRwLock<HashMap<Descriptor, Mutex<Peer>>>,
```

Comment:
Expand the comment to say when we _do_ hold a lock, and which. Right now, we only get a description of `peers` read/write lock requirements from `node_id_to_descriptor` comments.